### PR TITLE
Creating composable for connection monitoring and replacing existing logic with it

### LINKED ISSUE #12205: Create composable for connection monitoring and replace existing logic with it
Currently, connection monitoring is handled by [vuex state](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/state/modules/connection.js), this can be replaced by an internal facing (not exposed in the core API) composable module that defines state at the module level in order to provide a global record of connection monitoring.

### PR DESCRIPTION
**Pull request by nathanaelg16**
## Summary
This change creates a new composable for connection monitoring and migrates existing logic to use the new composable.

## References
- Fixes #12205

## Testing checklist

- [X] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/assets/src/composables/useConnection.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { ref } from 'kolibri.lib.vueCompositionApi';
+import Vue from 'vue';
+import VueCompositionApi from '@vue/composition-api';
+
+Vue.use(VueCompositionApi);
```
Comment (rtibbles) [inline kolibri/core/assets/src/composables/useConnection.js]: You shouldn't need to use the VueCompositionApi in the individual composable, as we do that on a global level, so I am fairly sure this should work without this.

**Review by rtibbles — COMMENTED** [review]
This looks like the correct approach, and the tests have been nicely updated to handle the new code. One small piece of cleanup, and I'll do some manual testing and closer review soon!

File: kolibri/core/assets/src/composables/useConnection.js (hunk 1 above)
Comment (nathanaelg16) [reply kolibri/core/assets/src/composables/useConnection.js]: I'm not sure why but I get failing tests if I don't include this. Are you able to reproduce this?

Comment (nathanaelg16) [reply kolibri/core/assets/src/composables/useConnection.js]: Fixed

**Review by rtibbles — APPROVED** [review]
Code changes look good, manual testing checks out. Snackbar properly displays in disconnected state, reconnect time does appropriate back off behaviour, and reconnection happens as intended.

**Comment by rtibbles** [comment]
Excellent work @nathanaelg16, very cleanly and neatly done!

**Pull request merged**